app/models/download.rb

```diff
@@ -1,22 +1,13 @@
 class Download
-  # Given a work and a format or mime type, generate a download file
-  def self.generate(work, options = {})
-    new(work, options).generate
-  end
-
-  # Remove all downloads for this work
-  def self.remove(work)
-    new(work).remove
-  end
-
-  attr_reader :work, :file_type, :mime_type
+  attr_reader :work, :file_type, :mime_type, :timestamp
 
   def initialize(work, options = {})
     @work = work
     @file_type = set_file_type(options.slice(:mime_type, :format))
     # TODO: Our current version of the mime-types gem doesn't include azw3, but
     # the gem cannot be updated without updating rest-client
     @mime_type = @file_type == "azw3" ? "application/x-mobi8-ebook" : MIME::Types.type_for(@file_type).first
+    @timestamp = Time.now.to_i
```

Comment:
Is this unique enough or should I go for [fractions of seconds](https://ruby-doc.org/core-2.3.4/Time.html#method-i-to_f) as well?
```
(Time.now.to_f * 1000).to_i
```

Reply:
If you're concerned about timing issues, what about using `Dir.mktmpdir`? It uses random numbers and checks for `EEXIST` on `mkdir`.

Reply:
Nice, thanks! I'll use that. I need to rewrite the tests to not rely on file location checks since we won't know the directory name.